client/core/core.go

```diff
@@ -6718,6 +6764,10 @@ func handleMatchProofMsg(c *Core, dc *dexConnection, msg *msgjson.Message) error
 		return fmt.Errorf("match proof note unmarshal error: %w", err)
 	}
 
+	// May need to reconcile epoch status orders from old epochs (missed
+	// preimage request and reconnected while order still in epoch status).
+	go checkMarketEpochOrders(dc, note.MarketID, note.Epoch)
```

Comment:
This bug is stranger.  You can end up with an order that stays in **epoch status** forever.

This is when you submit an order, lose connection, server sends a preimage request while you are gone, you come back before the end of preimage collection for that epoch (order is still epoch status according to server during reconciliation), but since it is a preimage miss, the order gets neither a `nomatch` or a `match` request from the server where the order status normally advances.  Unless you reconnect or restart to force another order status reconciliation, there is nothing to clean up these orphaned epoch status orders.

It's hard to test this, but you can manage the timing if you retry several times, usually with a CTRL+Z to suspend, and then wait a bit and resume (fd), hopefully after the request has failed but the server hasn't yet changed the order status to revoked yet.

The solution here is when new epochs are reported in the match_proof, we check all our epoch status trades to see if they are from a bygone epoch.

Reply:
No because it's exactly the problem that the reconnect happens before the order changes from epoch status to revoked server side.

Reply:
This issue of relying on the match_proof is troublesome.  There could be regular checks in tick, but the intervals would be larger and the check would be across all markets.  Alternatively `handleReconnect` could *schedule* a check for a couple epochs in the future, just to be safe.

Reply:
> Alternatively `handleReconnect` could _schedule_ a check for a couple epochs in the future, just to be safe.

That's kinda where my head was going too. On reconnect, we can mark all epoch orders for a check upon some conditions set of conditions in tick.

Reply:
I'll take a crack at that.  On reconnect, if there are any epoch status orders (where we might have missed the preimage request), schedule some `checkMarketEpochOrders` for the relevant markets...